Presentation.Web/app/components/data-processing/tabs/data-processing-agreement-tab-reference-create-modal.controller.ts

```diff
@@ -0,0 +1,87 @@
+﻿module Kitos.DataProcessing.Agreement.Edit.Reference.Create {
+    "use strict";
+
+    export class CreateReferenceDataProcessingAgreementController {
+        static $inject: Array<string> = [
+            "dataProcessingAgreementService",
+            "$scope",
+            "notify",
+            "$state",
+            "$stateParams",
+            "$uibModalInstance",
+            "referenceService"
+
+        ];
+
+        constructor(
+            private readonly dataProcessingAgreementService: Services.DataProcessing.IDataProcessingAgreementService,
+            private readonly $scope,
+            private readonly notify,
+            private readonly $state: angular.ui.IStateService,
+            private readonly $stateParams,
+            private readonly $uibModalInstance: ng.ui.bootstrap.IModalServiceInstance,
+            private referenceService: Services.ReferenceService) {
+
+            $scope.modalTitle = "Opret reference";
+        }
+
+        save() {
+
+            var msg = this.notify.addInfoMessage("Gemmer række", false);
+
+            this.referenceService.createReference(
+                this.$stateParams.id,
+                this.$scope.reference.title,
```

Comment:
assign this.$scope.reference til en var som du checker inden du kalder ned og så har du ikke det dybe hierarki gentaget så mange gange som her